http/handler.go

```diff
@@ -513,6 +516,35 @@ func WrapForwardedForHandler(h http.Handler, l *configutil.Listener) http.Handle
 		}
 
 		r.RemoteAddr = net.JoinHostPort(acc[indexToUse], port)
+
+		// Import the Client Certificate forwarded by the reverse proxy
+		// There should be only 1 instance of the header, but looping allows for more flexibility
+		clientCertHeaders, clientCertHeadersOK := r.Header[textproto.CanonicalMIMEHeaderKey(clientCertHeader)]
+		if clientCertHeadersOK && len(clientCertHeaders) > 0 {
+			var client_certs []*x509.Certificate
+			for _, header := range clientCertHeaders {
+				// Multiple certs should be comma delimetered
+				vals := strings.Split(header, ",")
+				for _, v := range vals {
+					decoded, err := url.QueryUnescape(v)
+					if err != nil {
+						respondError(w, http.StatusBadRequest, fmt.Errorf("failed to unescape the client certificate: %w", err))
+					}
+					block, err := base64.StdEncoding.DecodeString(decoded)
+					if block == nil {
+						respondError(w, http.StatusBadRequest, fmt.Errorf("failed to decode the client certificate: %w", err))
+						return
+					}
+					cert, err := x509.ParseCertificate(block)
+					if err != nil {
+						respondError(w, http.StatusBadRequest, fmt.Errorf("failed to parse the client certificate: %w", err))
+						return
+					}
+					client_certs = append(client_certs, cert)
+				}
+			}
+			r.TLS.PeerCertificates = client_certs
```

Comment:
Note that, if no TLS was used between the LB and us, `r.TLS` will be nil.

Reply:
Right, it does currently, but if we were to implement this, we wouldn't necessarily need to require it if the LB and Vault were colocated on the same machine (e.g., over localhost). This is a useful deployment scenario with Vault Enterprise's Seal Wrap FIPS compliance, wherein Vault's TLS isn't necessarily FIPS compliant, but putting a FIPS-validated LB in front of it would be useful.

Reply:
The condition I added in 7396430e21b30ae0d62d66a002ddaf268cca6e71 should keep it from panicking. It will just return an error now.

Allowing non-secured traffic between the LB and the server (even if it is on localhost) would open it up to a MITM attack or traffic sniffing. Returning an error instead of allowing a potential insecure setup would be the preferred option in my opinion. Especially in situations where FIPS is involved.

Reply:
> Allowing non-secured traffic between the LB and the server (even if it is on localhost) would open it up to a MITM attack or traffic sniffing.

Not sure this is true if traffic between the LB and server are [encrypted in transit](https://cloud.google.com/docs/security/encryption-in-transit). e.g. if Vault is running in Cloud Run behind a serverless LB.

Reply:
Same would also apply for unix: https://developer.hashicorp.com/vault/docs/configuration/listener/unix 

Having root on the box is sufficient to dump Vault's memory anyways, and so there's more interesting attack vectors than intercepting (plaintext) unix socket traffic.